Resources/public/js/views/services/plugins/ez-locationswapplugin.js

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-locationswapplugin', function (Y) {
+    "use strict";
+    /**
+     * Provides the plugin for swap location
+     *
+     * @module ez-locationswapplugin
+     */
+    Y.namespace('eZ.Plugin');
+
+    /**
+     * Swap location plugin. It sets an event handler to swap location
+     *
+     * In order to use it you need to fire `swapLocation` event with parameter
+     * `location` containing the eZ.Location object you want swap with other.
+     *
+     * @namespace eZ.Plugin
+     * @class LocationSwap
+     * @constructor
+     * @extends eZ.Plugin.ViewServiceBase
+     */
+    Y.eZ.Plugin.LocationSwap = Y.Base.create('locationswapplugin', Y.eZ.Plugin.ViewServiceBase, [], {
+
+        initializer: function () {
+            this.onHostEvent('*:swapLocation', this._selectDestinationLocation);
+        },
+
+        /**
+         * swapLocation event handler, launch the universal discovery widget
+         * to choose the location you want to swap with the given one
+         *
+         * @method _selectDestinationLocation
+         * @private
+         * @param {EventFacade} e swapLocation event facade
+         */
+        _selectDestinationLocation: function (e) {
+            console.log(e);
```

Comment:
yep... don't worry. it was just a try to see what i had in e... as said, i can't find the location from there...